### PR TITLE
webui: availability map update 5

### LINKED ISSUE #4275: Feature Request: Availability Map - Filter based on Device Groups, and custom Title
It would be nice if it was possible to filter results in the availability map to match the Device Groups.

It would also be really great if you could give the widget a custom title to match what set of devices it contained.

Thanks in advance.

### LINKED ISSUE #4345: Availability map widget issue
Hi
Availability map widget on the dashboard always shows devices with small icons. Like this:

Even if I change tile width (I tried different values)

But in Overview -> Maps -> Availability it looks like this:

Is it a bug? What is the use of the tile width?

### PR DESCRIPTION
**Pull request by crcro**
#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

update:
 - webui: can select detailed view (new view) direct page box size from global settings -> webui
 - code: added numeric field as input type (used for size input of the new webui changes)
 - db: cleaned webui db config names, enabled all help tips, added clearer tips
 - widget: removed the display option (got from session)
 - widget: added the custome title (fix #4275)
 - widget: select compact view (old view) tile size from widget settings (fix #4345)

last pr got somehow a lot of files as changed ...

### CODE REVIEW

**Comment by murrant** [comment]
Thanks for all the work.  I have a few questions (some may only be loosely related to this PR)

1. Don't call it Old availability map, call it Compact view or something.
2. Why are the settings in the global settings instead of right under the widget?  Resizing the squares for example now takes about 5 clicks and 3 page loads instead of 2 and 1.  (often you have to try multiple values to get the width right)
3. The "Old view" could need at least a 1px border, but I'm not sure, maybe it is better without.
4. It would be neat to click on a total item and filter the display (not sure that makes sense in the widget)

**Comment by crcro** [comment]
1. Ok
2. I was thinking that is more like "clean", all settings in one place, not some settings here and some others there
3. No border, no margin etc was the desired output from users 
4. I don't understand this ...

**Comment by murrant** [comment]
2. Yes, but in this case, I think the cost is too high.
3. Sounds good
4.  I'll make a new issue.

**Comment by laf** [comment]
I think I'd say that the settings should be in the widget like murrant has said, this allows per user customisation rather than globally.

**Comment by crcro** [comment]
ok, will change

File: html/includes/common/availability-map.inc.php [hunk 1]
```diff
@@ -12,41 +12,58 @@
  * the source code distribution for details.
  */
 
+$sql = dbFetchRow('SELECT `settings` FROM `users_widgets` WHERE `user_id` = ? AND `widget_id` = ?', array($_SESSION["user_id"], '1'));
+$widget_mode = json_decode($sql['settings']);
```
Comment (laf) [inline html/includes/common/availability-map.inc.php]: We should probably pass true to json_decode here to keep in line with how it's done elsewhere.

File: html/pages/settings/webui.inc.php [hunk 2]
```diff
@@ -31,6 +31,10 @@
           'descr'              => 'Use device groups filter',
           'type'               => 'checkbox',
     ),
+    array('name'               => 'webui.availability_map_box_size',
```
Comment (laf) [inline html/pages/settings/webui.inc.php]: This probably should just be moved to the widget settings if you can, allows each user to fully customise them.

Comment (crcro) [inline html/pages/settings/webui.inc.php]: it is for the direct page access, not for widget

Comment (laf) [inline html/pages/settings/webui.inc.php]: Ah sorry, leave as is then.

**Comment by laf** [comment]
Just the comment about json decode.

I don't see any config options other than Widget title in the dashboard.

**Comment by crcro** [comment]
There is one more option for tile width if compact (old view) is selected.

File: html/includes/common/availability-map.inc.php (hunk 1 above)
Comment (jonathon-k) [inline html/includes/common/availability-map.inc.php]: @crcro This query creates the potential for unexpected behavior (i.e. if you have this widget on multiple dashboards, with different settings, you don't know which one you will get) and defeats the dashboard specific query for widget settings that is already being done.  Which also means you don't know which dashboard will control the direct page view.  Can we rework this so the widget and direct view are independent entities?

**Comment by crcro** [comment]
@jonathon-k … yes it is possible, i will submit a fix tommorow hopefully

**Pull request merged**